pkg/providers/vsphere/govmomiclient.go

```diff
@@ -0,0 +1,107 @@
+package vsphere
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type VSphereClient interface {
+	Username() string
+	GetPrivsOnEntity(ctx context.Context, path string, objType string, username string) ([]string, error)
+}
+
+type VMOMIClient struct {
+	Gcvm     *govmomi.Client
+	Finder   *find.Finder
+	username string
+}
+
+type VSphereClientBuilder interface {
+	Build(ctx context.Context, host string, username string, password string, insecure bool, datacenter string) (VSphereClient, error)
+}
+
+type VMOMIClientBuilder struct{}
+
+func (*VMOMIClientBuilder) Build(ctx context.Context, host string, username string, password string, insecure bool, datacenter string) (VSphereClient, error) {
+	u, err := soap.ParseURL(host)
+	u.User = url.UserPassword(username, password)
+	if err != nil {
+		return nil, err
+	}
+
+	// start gvmc
+	gvmc, err := govmomi.NewClient(ctx, u, insecure)
+	if err != nil {
+		return nil, err
+	}
+	err = gvmc.Logout(ctx)
+	if err != nil {
+		return nil, err
+	}
+	err = gvmc.Login(ctx, u.User)
+	if err != nil {
+		return nil, err
+	}
+
+	f := find.NewFinder(gvmc.Client, true)
+
+	dc, err := f.Datacenter(ctx, datacenter)
+
+	f.SetDatacenter(dc)
+
+	vsc := VMOMIClient{gvmc, f, username}
+
+	return &vsc, err
+}
+
+func (vsc *VMOMIClient) Username() string {
+	return vsc.username
+}
+
+func (vsc *VMOMIClient) GetPrivsOnEntity(ctx context.Context, path string, objType string, username string) ([]string, error) {
+	var objRef types.ManagedObjectReference
```

Comment:
What does this object ref refer to?

Reply:
This will be a ref to an object in vsphere. I've tried renaming it, let me know if you think it's clearer?

Reply:
What is the official name that vcenter uses? I might like `resource` more otherwise

Reply:
I don't have a strong opinion here and am happy to go with whatever you think is best.

In general, I really don't like "object" for variable names because it's so vague. At the same time, golang's naming conventions go against a lot of what I'm used to, so I've been tossing my personal preferences out the window lately. 😅

I unfortunately do see "object" in a lot of the docs, but again am happy to go with whatever you think is best:
https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vcenterhost.doc/GUID-031BDB12-D3B2-4E2D-80E6-604F304B4D0C.html

https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vcenterhost.doc/GUID-7FDFBDBE-F8AC-4D00-AE5E-3F14D7472FAF.html

https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vcenterhost.doc/GUID-E8E854DD-AA97-4E0C-8419-CE84F93C4058.html

I am guessing that these persistent references to "objects" in the govmomi code derive from the "object" language in the vsphere docs.